sklego/feature_selection/mrmr.py

```diff
@@ -0,0 +1,216 @@
+import warnings
+
+import numpy as np
+from sklearn.base import BaseEstimator
+from sklearn.feature_selection import f_classif, f_regression
+from sklearn.feature_selection._base import SelectorMixin
+from sklearn.utils.validation import check_is_fitted, check_X_y
+
+
+def _redundancy_pearson(X, selected, left):
+    """Redundancy function for the MRMR feature selector algorithm
+
+    Parameters
+    ----------
+    X : array-like, shape=(n_samples, n_features,)
+        Training data. Used to compute redundancy of the training features.
+    selected : array-like.
+        List of indexes of the selected features at iteration i-th.
+    left : array-like.
+        List of indexes of the left features at iteration i-th. Mrmr will select a feature
+        from this list.
+
+    Returns
+    -------
+    np.ndarray, shape = (len(left), )
+        The array containing the redundancy score using pearson correlation.
+    """
+    # if len(selected) == 0:
+    #     return np.ones(len(left))
+
+    X_norm = X - np.mean(X, axis=0, keepdims=True)
+    Xs = X_norm[:, selected]
+    Xl = X_norm[:, left]
+
+    num = (Xl[:, None, :] * Xs[:, :, None]).sum(axis=0)
+    den = np.sqrt((Xl[:, None, :] ** 2).sum(axis=0)) * np.sqrt((Xs[:, :, None] ** 2).sum(axis=0))
+
+    return np.sum(np.abs(np.nan_to_num(num / den, nan=np.finfo(float).eps)), axis=0)
+
+
+class MaximumRelevanceMinimumRedundancy(SelectorMixin, BaseEstimator):
+    """Maximum Relevance Minimum Redundancy (MRMR) is an iterative feature selection method commonly used in data
```

Comment:
I would add the formula in the docstring as well, something along the line:

$\text{score}_{i}(f) = \frac{\text{relevance}(f | y}{\text{reduncancy}(f | \text{selected}_{i-1}}$
for each feature $f$ is

Edit: Not sure why github doesn't like the formula :|

Reply:
Maybe this? 

$` \text{score}_{i}(f) = \frac{\text{relevance}(f | y)}{\text{reduncancy}(f | \text{selected}_{i-1})} `$